### ISSUE TITLE
Negative sobol indices

### ISSUE THREAD

**Issue by jennywang123** [SALib/SALib#135]
In the example problem in the Salib user manual, the S1 is [ 0.30644324 0.44776661 -0.00104936 ]. Sobol indices should not be negative as it is based on variance decomposition. Is this a bug? Thanks!

**Comment by willu47** [comment]
See #102

**Comment by willu47** [comment]
Following this up, the documentation page [getting started](http://salib.readthedocs.io/en/latest/getting-started.html) lists the first-order sensitivity indices from the sobol example as `[0.30644324, 0.44776661, -0.00104936]`.

Using v1.0.3, I get the following results: `[0.30797549,  0.44776661, -0.00425452]` so, a different number and still negative.

**Comment by calvinwhealton** [comment]
The version of the first-order index estimator used in SALib is not guaranteed to be positive. Only in expectation will it be positive and we have sample estimates. The Saltelli et al (2010, Table 2, Equation b) estimator uses vectors of model values evaluated at matrices A, B, and A_B. There is no guarantee that \sum[ f(B)*(f(A_B) - f(A))] is always positive.

Saltelli et al (2010, Table 2, Equation c) might be less prone to this problem because it has a sum that is guaranteed to be positive, but it is subtracted from an estimate of the variance of the model. For small values of the S1 index this formula could also be negative due to sampling variability.

**Comment by jdherman** [comment]
This is an important question -- as Calvin said, the estimators can sometimes be negative. I would expect the negative values to always be relatively small, and with confidence intervals that overlap zero (i.e., they are statistically indistinguishable from zero).

If you ever run into a case where the negative SI value is larger, and the confidence interval does not overlap zero, let us know. But I'll close the issue for now.

**Comment by baherehvojdani** [comment]
Hello 
I have an problem, in my research , for all x that I have, the first-order is negative and Total orde is more than 0.3. Is something wrong?

**Comment by ConnectedSystems** [comment]
Hi @baherehvojdani 

We'd need to know the following before we can give any help:

1. Which method are you using?
2. How many parameters?
3. How many samples?

**Comment by baherehvojdani** [comment]
hello dear
tanks for answering, my model is SVR and i have 4 independent variables and 1 dependent variable .

**Comment by ConnectedSystems** [comment]
Hi @baherehvojdani 

Briefly, 305 samples is probably not enough.

Instead of responding here, could you please open a [new issue here](https://github.com/SALib/SALib/issues/new) to avoid spamming everyone else?

**Comment by MAGALA-RICHARD** [comment]
I would recommend increasing the sampling factor and hence adjusting the sample space
